include/common/mpmc_queue.h

```diff
@@ -70,6 +51,28 @@ class mpmc_queue{
          internal_pread = 0;
          internal_pwrite = 0;
       }
+
+      bool empty();
+      T pop();
+      bool push(T item);
+
+   private:
+      bool full(unsigned long long current);
+      bool empty(unsigned long long current);
+
+      int size;
+      std::vector<T> buffer;
+      std::atomic<unsigned long long> pread;
+      std::atomic<unsigned long long> pwrite;
+      std::atomic<unsigned long long> internal_pread;
+      std::atomic<unsigned long long> internal_pwrite;
+
+      bool lockfree = false;
+
+      std::mutex mutex;
```

Comment:
A better name for mutex? It is a bad idea to give the same name to a variable than and std:: symbol